clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/validators/BasicAmazonDeployDescriptionValidator.groovy

```diff
@@ -73,6 +76,37 @@ class BasicAmazonDeployDescriptionValidator extends AmazonDescriptionValidationS
     if (!description.source?.useSourceCapacity) {
       validateCapacity description, errors
     }
+
+    // unlimitedCpuCredits (set to true / false) is valid only with supported instance types
+    if (description.unlimitedCpuCredits != null && !InstanceTypeUtils.isBurstingSupported(description.instanceType)) {
+      errors.rejectValue "unlimitedCpuCredits", "basicAmazonDeployDescription.bursting.not.supported.by.instanceType"
+    }
+
+    logWarnings(description)
+  }
+
+  /**
+   * Log warnings to indicate potential user error.
+   */
+  private void logWarnings(BasicAmazonDeployDescription description) {
+
+    // certain features work as expected only when AWS EC2 Launch Template feature is enabled and used
+    if (!description.setLaunchTemplate) {
+      def ltFeaturesEnabled = []
+      if (description.requireIMDSv2) {
+        ltFeaturesEnabled.add("requireIMDSv2")
+      }
+      if (description.associateIPv6Address) {
+        ltFeaturesEnabled.add("associateIPv6Address")
+      }
+      if (description.unlimitedCpuCredits != null) {
+        ltFeaturesEnabled.add("unlimitedCpuCredits")
+      }
+
+      if (ltFeaturesEnabled) {
+        log.warn("WARNING: The following fields {} work as expected only with AWS EC2 Launch Template, but 'setLaunchTemplate' is set to false.", ltFeaturesEnabled)
```

Comment:
this is handy, thanks

Reply:
FYI, added a little bit of information about the request to this warning, in latest changes.